go/vt/automation/scheduler_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package automation
+
+import (
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	context "golang.org/x/net/context"
+
+	pb "github.com/youtube/vitess/go/vt/proto/automation"
+)
+
+func TestSchedulerImmediateShutdown(t *testing.T) {
+	// Make sure that the scheduler shuts down cleanly when it was instantiated, but not started with Run().
+	scheduler, err := NewScheduler()
+	if err != nil {
+		t.Fatalf("Failed to create scheduler: %v", err)
+	}
+	scheduler.ShutdownAndWait()
+}
+
+func TestEnqueueSingleTask(t *testing.T) {
+	enqueueClusterOperationAndCheckOutput(t, "TestingEchoTask", "echoed text")
+}
+
+func TestEnqueueEmittingTask(t *testing.T) {
+	enqueueClusterOperationAndCheckOutput(t, "TestingEmitEchoTask", "echoed text from emitted task")
+}
+
+func enqueueClusterOperationAndCheckOutput(t *testing.T, taskName string, expectedOutput string) {
+	scheduler, err := NewScheduler()
+	if err != nil {
+		t.Fatalf("Failed to create scheduler: %v", err)
+	}
+	scheduler.setTaskCreator(testingTaskCreator)
```

Comment:
It might makes sense to pull this code out into a function, otherwise I can see it being really easy to forget to set the task creator.

Instead: 

`scheduler := newTestScheduler()`